model/project_vars.go

```diff
@@ -556,24 +576,104 @@ func isParameterStoreEnabledForProject(ctx context.Context, projectID string) (b
 		return false, nil
 	}
 
-	projRef, err := FindMergedProjectRef(projectID, "", false)
+	return ref.ParameterStoreEnabled, nil
+}
+
+// findProjectRef finds the project ref associated with the project variables.
+// Returns a bool indicating if it's a branch project ref or a repo ref.
+func (projectVars *ProjectVars) findProjectRef() (ref *ProjectRef, isRepoRef bool, err error) {
+	projectID := projectVars.Id
+	// This intentionally looks for a branch project ref without merging with
+	// its repo ref because project vars for a branch project are stored
+	// separately from project vars for a repo. Therefore, a branch project and
+	// its repo could have differing sync statuses (e.g. it's possible for a
+	// branch project's vars to be synced to Parameter Store, but not its repo
+	// vars).
+	projRef, err := FindBranchProjectRef(projectID)
 	if err != nil {
-		return false, errors.Wrapf(err, "finding merged project ref '%s'", projectID)
+		return nil, false, errors.Wrapf(err, "finding merged project ref '%s'", projectID)
 	}
 	if projRef != nil {
-		return projRef.ParameterStoreEnabled, nil
+		return projRef, false, nil
 	}
 
 	// Project vars could tied to a repo instead of branch project, so check the
 	// repo as a fallback.
 	repoRef, err := FindOneRepoRef(projectID)
 	if err != nil {
-		return false, errors.Wrapf(err, "finding repo ref '%s'", projectID)
+		return nil, false, errors.Wrapf(err, "finding repo ref '%s'", projectID)
 	}
 	if repoRef == nil {
-		return false, errors.Errorf("project or repo ref '%s' not found", projectID)
+		return nil, false, errors.Errorf("project or repo ref '%s' not found", projectID)
+	}
+	return &repoRef.ProjectRef, true, nil
+}
+
+// TODO (DEVPROD-11882): remove full sync logic once the Parameter Store
+// rollout is complete. This functionality only exists to aid the migration
+// process.
+func fullSyncToParameterStore(ctx context.Context, vars *ProjectVars, pRef *ProjectRef, isRepoRef bool) error {
+	grip.Debug(message.Fields{
+		"message":                     "fully syncing project vars to Parameter Store",
+		"num_vars":                    len(vars.Vars),
+		"existing_parameter_mappings": vars.Parameters,
```

Comment:
Is it important to log this? I'm guessing it can be pretty long for some projects or is just the  num_vars enough for debugging purposes?

Reply:
It's important for debugging while the migration is ongoing so that I can keep track of which projects get synced/when. Just `num_vars` is enough for me to have the info I need to track it.